---
title: "KD processed data analysis"
author: "EZ"
date-modified: "Last compiled on `r format(Sys.time(), '%d %B, %Y')`" 
format: 
  html:
    html-math-method: katex
    toc: true
    toc-location: left
    toc-depth: 5
    toc_float:
      collapsed: true
      smooth_scroll: true
    code-fold: true  
    highlight-style: a11y-dark
    default-image-extension: svg
theme:
  light: flatly
  dark: darkly
execute:
  warning: false
editor: visual
---

## Context

This document compiles possible data visualizations made using processed data received from Soilytex.

## 16S rRNA gene

The 16S rRNA gene has been a mainstay of sequence-based bacterial analysis for decades. The \~1500â€‰bp 16S rRNA gene comprises nine variable regions throughout the highly conserved 16S sequence. I do not have the information if soilytix uses 3^d^ generation sequencing (*i.e* long read for full 16S) which would lead to the best taxonomic resolution ([Johnson et al. 2019](https://rdcu.be/edQHU)).

### Relative abundance data

#### Species:

I am assuming that relative abundances are expressed in percentages per Sample-ID and per sequenced region (here 16S).

```{r}

#| label: loading data
#| results: hide
#| warning: false

library("tidyverse")
library("kableExtra")

map <- read_tsv("data/mapfile2.txt") %>% 
  mutate(Sample_ID = as.factor(Sample_ID))

species <- read_csv2("data/16S_Abundance_Species.csv") %>% 
  mutate(Sample_ID = as.factor(Sample_ID)) %>% 
  pivot_longer(!Sample_ID, names_to = "bacteria_species", values_to = "RA_bacteria") %>% 
  left_join(y = map, by = "Sample_ID" ) %>% 
  mutate_if(is.character,as.factor) %>% 
  arrange(desc(RA_bacteria))


# length(unique(species$bacteria_species))
# max(species$RA_bacteria)
```

```{r}
#| warning: false
#| echo: false
#| label: table species
#| results: false
kbl(species[1:300,]) %>%
  kable_paper() %>%
  column_spec(2, italic = TRUE) %>% 
  scroll_box(width = "100%", height = "300px")
```

there are 321 identified bacteria species. the maximum relative abundance calculated is \~9% whereas the median is 0 and the mean is \~1%.

Here, I will focus on the top 5 identified species per sample

```{r}
#| label: Top 5 identified bacteria species per sample
#| results: hide
#| warning: false

top5 <- species %>% 
  group_by(Sample_ID) %>% 
  slice_max(order_by = RA_bacteria, n = 5) %>% 
  ungroup() %>% 
  arrange(desc(RA_bacteria)) %>% 
  drop_na() %>% 
  mutate(Miete = fct_relevel(Miete, c("T1", "T4", "T6", "T7", "T8", "T9","T10", "T11")))

top5_bact <- unique(top5$bacteria_species)
length(top5_bact)
```

There are 41 different species that across the top 5 identified species per sample. The median relative abundance is \~0.8 %. When eliminating NA values, we're left with 35 species.

```{r}
#| warning: false
#| echo: false
#| label: table top 5 species
kbl(top5) %>%
  kable_paper() %>%
  column_spec(2, italic = TRUE) %>% 
  scroll_box(width = "100%", height = "300px")
```

```{r }
#| label: heatmap species
#| echo: false
#| warning: false
#| out-height: 100%
#| out-width: 100%
#| output: True
#| results: false
#| dev: svg

library("scales")
library("RColorBrewer")
library("ggtext")
library("glue")
library("ggpubr")
library("ggh4x")

# color -------------------------------------------------------------------
gradient_base <- c("#f2edee")

grad <- function(x) colorRampPalette(c(gradient_base,x))(200)

# grad(c("#380282"))


# plot --------------------------------------------------------------------

p <- ggplot(data = top5,
                  mapping = aes( x = Zeit_Monate,
                                 y= bacteria_species,
                                 fill = RA_bacteria)) +
  geom_tile() +
  scale_fill_gradientn(name = "relative abundance bacteria (%)",colours =c("#f2edee","orange", "#380282") ,
                       na.value = "grey55",
                       values = rescale(x = c(0, 2.5, 10), from = c(0, 10)),limits=c(0,10))+ 
  facet_grid2(~ Miete ,
             labeller = label_wrap_gen(multi_line = TRUE),
             switch = "y",
             space = "free",
             scales = "free") +
  # xlab(label = "time (months)") +
  labs(
    caption = "Heatmap top 5 specie with Zeit - Miete interaction"
  )+
  theme_minimal() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        axis.text.y = element_text(size = 8, face = "bold.italic"),
        axis.text.x = element_text(size = 10, face = "bold", hjust = 0, vjust= 0.5, angle = 90),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        strip.text.y.left = element_text(size= 7, face = "bold", angle = 0),
        strip.background.x = element_rect (colour = c("#890100"),
                                          linetype = 1,
                                          linewidth = 1),
        # ggh4x.facet.nestline = element_line(colour = c("#380282")),
        strip.text.x = element_text(size= 8, face = "bold"),
        legend.title = element_text(face = "bold", 
                                    size = 8,
                                    margin = margin(r = 10)),
        legend.text = element_text(size = 8),
        legend.position = "bottom" ) +
  guides(fill = guide_colorbar(barwidth = 20, barheight = 0.8, title.vjust = 1))
p                                      

```

```{r}
#| label: heatmap species2
#| echo: false
#| warning: false
#| results: false
#| output: True
#| out-height: 100%
#| out-width: 100%
#| dev: svg



library("scales")
library("RColorBrewer")
library("ggtext")
library("glue")
library("ggpubr")
library("ggh4x")

# color -------------------------------------------------------------------
gradient_base <- c("#f2edee")

grad <- function(x) colorRampPalette(c(gradient_base,x))(200)

# grad(c("#380282"))


# plot --------------------------------------------------------------------

p <- ggplot(data = top5,
                  mapping = aes( x = Sample_ID,
                                 y= bacteria_species,
                                 fill = RA_bacteria)) +
  geom_tile() +
  scale_fill_gradientn(name = "relative abundance bacteria (%)",colours =c("#f2edee","orange", "#380282") ,
                       na.value = "grey55",
                       values = rescale(x = c(0, 2.5, 10), from = c(0, 10)),limits=c(0,10))+ 
  facet_grid2(~ Zeit_Monate ,
             labeller = label_wrap_gen(multi_line = TRUE),
             switch = "y",
             space = "free",
             scales = "free") +
  # xlab(label = "time (months)") +
  theme_minimal() +
  labs(
    caption = "Heatmap top 5 species through time"
  )+
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        axis.text.y = element_text(size = 8, 
                                   face = "bold.italic"),
        axis.text.x = element_text(size = 10, 
                                   face = "bold", 
                                   hjust = 0, 
                                   vjust= 0.5, 
                                   angle = 90),
        axis.title.x = element_blank(),
        strip.background = element_blank(),
        strip.text.y.left = element_text(size= 7, 
                                         face = "bold", angle = 0),
        strip.background.x = element_rect (colour = c("#890100"),
                                          linetype = 1,
                                          linewidth = 1),
        # ggh4x.facet.nestline = element_line(colour = c("#380282")),
        strip.text.x = element_text(size= 8, face = "bold"),
        legend.title = element_text(face = "bold", 
                                    size = 8,
                                    margin = margin(r = 10)),
        legend.text = element_text(size = 8),
        legend.position = "bottom" ) +
  guides(fill = guide_colorbar(barwidth = 20, barheight = 0.8, title.vjust = 1))
p                                      

```

#### Class:

We can also visualize the relative abundance of the most abundant classes.

```{r}
#| label: Top classes per sample
#| results: hide
#| warning: false

library("Hmisc")

class <- read_csv2("data/16S_Abundance_class.csv") %>% 
  mutate(Sample_ID = as.factor(Sample_ID)) %>% 
  mutate_if(is.character, as.factor) %>% 
  pivot_longer(!Sample_ID, names_to = "bacteria_class", values_to = "RA_bacteria") %>% 
  left_join(y = map, by = "Sample_ID" ) %>% 
  arrange(desc(RA_bacteria))
  # drop_na() %>%
  # mutate(Miete = fct_relevel(Miete, c("T1", "T4", "T6", "T7", "T8", "T9","T10", "T11")))

class$Miete <- NULL

classes_to_keep <- unique((class %>% filter(RA_bacteria>5))$bacteria_class)


class2 <- class %>%
  mutate(bacteria_class, 
         bacteria_class = if_else(bacteria_class %nin% classes_to_keep, str_c("Others <5%"), bacteria_class)) %>%
  mutate_if(is.character, as.factor) %>% 
  distinct() 

# the multiple copies of classes that are now labelled "Others <5%" produce a distorted effect when plotting
# to avoid that we'll sum the RA of all "Others <5%" for each condition and replace the multiple 
# entries by a single one

c_sum <- class2 %>% 
  filter(bacteria_class %in% "Others <5%") %>%
  group_by(Sample_ID) %>%
  mutate(RA_bacteria= sum(RA_bacteria)) %>%
  distinct() 

class2 <- class2 %>% 
  filter(bacteria_class %nin% "Others <5%") %>%
  bind_rows(c_sum)
```

There are 13 classes with relative abundances \>5% summarized in the table below.

```{r}
#| warning: false
#| echo: false
#| label: table top classes bacteria
kbl(class2 %>% 
      mutate(
        RA_bacteria = round(RA_bacteria, digits = 1)
        )
    ) %>%
  kable_paper() %>% 
  scroll_box(width = "100%", height = "300px")
```

```{r}
#| label: barplot classes
#| echo: false
#| warning: false
#| results: false
#| output: True
#| out-height: 100%
#| out-width: 100%
#| dev: svg

ct <- c( "#399283", "#84eb86", "#3e2690", "#7ce5e6", "#173d61", "#b4b7d4", "#71114b", "#59a20c", "#dd3dca", "#18441b", "#fbacf6", "#5577b2", "#c9dd87", "#5a66f9")

plot <-ggplot(class2, 
              aes(x= Sample_ID,
                  y= RA_bacteria,
                  group = Sample_ID,
                  fill=fct_reorder(bacteria_class, RA_bacteria)))+
  geom_bar(stat="identity", 
           position="stack") +
  scale_fill_manual(values= rev(ct))+
  
  # scale_y_continuous(labels=percent, expand = c(0,0))+
  theme_minimal()+
  labs(fill = "Class Bacteria")+
  ylab("Relative Abundance (%)")+
  facet_grid2 (~Zeit_Monate,
               scales="free_x",
               space = "free_x")+
  theme(strip.text.x = element_text(size = 12, 
                                    face="bold", 
                                    angle = 90, 
                                    vjust = 0.5,
                                    hjust = 0),
        strip.text.y = element_text(size = 10, 
                                    face="bold", 
                                    angle = 0),
        legend.title = element_text(size = 14, 
                                    face = "bold"), 
        legend.text = element_text(size = 12, 
                                   face = "bold"),
        axis.title.y = element_text(size= 12,
                                    face = "bold"),
        axis.text.y = element_text(size = 10, 
                                   face = "bold"),
        axis.text.x = element_text(size = 10, 
                                   face = "bold",
                                   hjust = 0.5, 
                                   vjust= 0.5,
                                   margin = margin(t = 1)),
        axis.title.x = element_text(size= 12,
                                    face = "bold",
                                    margin = 
                                      margin(t = 8)),
        panel.grid.major = element_line(colour = NA),
        panel.grid.minor = element_line(colour = NA))


plot

```

![](bacteria%20classes.svg){width="18.6cm"}

It seems as if the sum of classes doesn't add up to 100 (after checking their table, yes it doesn't). I am assuming this is because they delete unassigned ASVs/OTUs on each taxonomic rank.

### Alpha diversity indexes bacteria

Here I am just putting an example of visualization of the Shannon diversity index. The richness and evenness exact indexes weren't specified.

The Shannon Diversity Index (or Shannon-Weaver Index) is a way to measure the diversity of species in a community.

Denoted as H, this index is calculated as $H = \sum{p_i}*\ln({p_i})$ where ${p_i}$ is the proportion of the entire community made up of species *i*.

The higher the value of H, the higher the diversity of species in a particular community. The lower the value of H, the lower the diversity. A value of H = 0 indicates a community that only has one species.

```{r}
#| label: alpha div prep 
#| results: hide
#| warning: false


alpha_div <- read.csv2("data/alpha_div.csv") %>% 
  mutate(Sample_ID = as.factor(Sample_ID)) %>% 
  left_join(y = map, by = "Sample_ID" ) %>% 
  mutate_if(is.character, as.factor) 


alpha_div$Miete <- NULL

alpha_div <- drop_na(alpha_div)

combo <- combn(levels(alpha_div$Zeit_Monate), 2)

my_comparisons <- as.data.frame(combo) %>%
  sapply( function(x) as.character(x), simplify = FALSE)

library(ggpubr)

low <- function(x){mean(x)-sd(x)}

high <- function(x){mean(x)+sd(x)}
```

```{r}
#| label: Shannon diversity index 
#| results: hide
#| echo: false
#| output: false
#| warning: false
#| dev: svg

p_shannon1 <- ggplot(data = alpha_div , 
                    aes (x = Zeit_Monate,
                         group = Zeit_Monate,
                         color = Zeit_Monate,
                         label = Sample_ID,
                         y= Shannon)) +
  
  geom_point(size= 2, key_glyph = "rect")+
  geom_text(check_overlap = T, 
            size = 4, 
            hjust = 0, 
            nudge_x = 0.2) +
  stat_summary(fun="mean", 
               fun.min = low, 
               fun.max = high ,
               size=1, 
               shape= 4)+
  
  scale_color_manual(values = c("#72e5ef", "#fd1e6e", "#7191ce", "#79eb99", "#9c6f08", "#ddb35b", "#513886"),
                     name = "Zeit")+
  
  # facet_wrap(~location)+
  
  xlab("")+
  ylab("Shannon diversity index")+
  theme_classic()+
  theme(axis.title.y = element_text(size = 12, 
                                    margin =margin(t=0,r=8,b=0,l=0)),
        axis.title.x = element_text(size = 12,  
                                    margin =margin(t=8,r=0,b=0,l=0)),
        axis.text.x = element_text(size = 12, 
                                   colour = "black",
                                   angle = 0, 
                                   vjust = 0.5, 
                                   hjust = 0.5),
        axis.text.y = element_text(size = 10, 
                                   colour = "black"),
        legend.text  = element_text(size = 11),
        legend.title = element_text(size = 11, 
                                    face = "bold"),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, 
                                  face = "bold"))+
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     aes(label = after_stat(p.signif))) # Add pairwise comparisons p-value
  # stat_compare_means(method = "kruskal.test",
  #                    label.y = 11 , label.x = 1) + 
  # guides(
  # color = guide_legend(
  #   label.position = "top",
  #   override.aes = list(shape = 20, size = 1)
  # ))

# p_shannon1


p_shannon2 <- ggplot(data = alpha_div , 
                    aes (x = Zeit_Monate,
                         group = Zeit_Monate,
                         color = Zeit_Monate,
                         label = Sample_ID,
                         y= Shannon)) +
  
  geom_point(size= 2, key_glyph = "rect")+
  geom_text(check_overlap = T, 
            size = 4,
            nudge_x = .2,
            hjust = 0) +
  stat_summary(fun="mean", 
               fun.min = low, 
               fun.max = high ,
               size=1, 
               shape= 4)+
  
  scale_color_manual(values = c("#72e5ef", "#fd1e6e", "#7191ce", "#79eb99", "#9c6f08", "#ddb35b", "#513886"),
                     name = "Zeit")+
  
  # facet_wrap(~location)+
  
  xlab("")+
  ylab("Shannon diversity index")+
  theme_classic()+
  theme(axis.title.y = element_text(size = 12, 
                                    margin =margin(t=0,r=8,b=0,l=0)),
        axis.title.x = element_text(size = 12,  
                                    margin =margin(t=8,r=0,b=0,l=0)),
        axis.text.x = element_text(size = 12, 
                                   colour = "black",
                                   angle = 0, 
                                   vjust = 0.5, 
                                   hjust = 0.5),
        axis.text.y = element_text(size = 10, 
                                   colour = "black"),
        legend.text  = element_text(size = 11),
        legend.title = element_text(size = 11, 
                                    face = "bold"),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, 
                                  face = "bold"))+
  # stat_compare_means(comparisons = my_comparisons,
  #                    method = "wilcox.test",
  #                    aes(label = after_stat(p.signif))) +# Add pairwise comparisons p-value
  stat_compare_means(method = "kruskal.test" ,
                     label.x = 2,
                      ) + 
  guides(
  color = guide_legend(
    label.position = "top",
    override.aes = list(shape = 20, size = 1)
  ))

# p_shannon2


```

```{r}

#| label: Shannon diversity index tables 
#| results: hide
#| echo: false
#| output: true
#| warning: false
#| out-height: 100%
#| out-width: 100%
#| dev: svg


# ggpubr::compare_means(data = alpha_div,
#                       formula = Shannon ~ Zeit_Monate,
#                       method = "kruskal.test",
#                       p.adjust.method = "BH",
#                       paired = T)

pl <- 
  ggpubr::ggarrange(p_shannon1,
                  p_shannon2,
                  common.legend = F,
                  legend = "none",
                  ncol = 2,
                  labels = c("A)", "B)")
                  )
cap <- expression("Shannon diversity index values. \nThe samples are grouped by time the X marks the mean. The line represents the standard deviation. \nIn A) a pairwise comparison via Wilcoxon test is represented. \nIn B) only a global Kruskall-Wallis test is displayed. \nns = not significant")

annotate_figure(
        annotate_figure(pl,
                bottom =text_grob(paste(cap)),
        )
)
```

### Beta diversity

the values provided are coordinates on two axes without mentioning which data, which distance or which ordination were used.

So, if I just plot what's in the file that's what I would get.

```{r}
#| label: beta div
#| results: hide
#| echo: false
#| output: True
#| warning: false
#| out-width: 90%
#| out-height: 90%
#| dev: svg

beta <- read_csv2("data/beta_div_16s.csv") %>% 
  mutate(Sample_ID = as.factor(Sample_ID)) %>% 
  left_join(y = map, by = "Sample_ID" ) %>% 
  mutate_if(is.character, as.factor) 
beta$Miete <- NULL

col24 <- c("#52ef99", "#860967", "#b3e61c", "#1932bf", "#c7dd91", "#dc35dc", "#0ca82e", "#fe1d66", "#51f310", "#e68dd9", "#0c5f31", "#f17a74", "#41c9dc", "#a63838", "#2480a1", "#fe5900", "#523d6e", "#fabd2a", "#6524ff", "#7b9b47", "#846dff", "#f4bb8f", "#653d28", "#afc6fe")


beta2 <- drop_na(beta)

b <- ggplot(data = beta2,
            aes(x = PCo1,
                y = PCo2,
                shape = Zeit_Monate,
                colour = Sample_ID))+
  geom_point(size = 3)+
  scale_color_manual(values = col24)+
  scale_shape_manual(values= c(18,17,7,4,16,3,8))+
  theme_classic()+
  theme(axis.title.y = element_text(size = 12, 
                                    margin =margin(t=0,r=8,b=0,l=0)),
        axis.title.x = element_text(size = 12,  
                                    margin =margin(t=8,r=0,b=0,l=0)),
        axis.text.x = element_text(size = 12, 
                                   colour = "black",
                                   angle = 0, 
                                   vjust = 0.5, 
                                   hjust = 0.5),
        axis.text.y = element_text(size = 10, 
                                   colour = "black"),
        legend.text  = element_text(size = 11),
        legend.title = element_text(size = 11, 
                                    face = "bold"),
        strip.background = element_blank(),
        strip.text = element_text(size = 12, 
                                  face = "bold"))



b
```

Usually axis 1 / axis x or PCo1 is the dimension that explains the most variance in the dataset even if it is unknown exactly here. What we can see mostly here is that the microbial community is discriminated by the variable time mostly. on that axis from right to left, one can follow a progression mainly in the 3 first months. after that, the influence of time seems to diminish. I don't have enough knowledge about the samples to be able to guess what might be the variables behind the scatter of the samples.
